simulator/examples/bmp.rs

```diff
@@ -2,22 +2,18 @@
 //!
 //! This example uses `impl From<Rgb565> for SimPixelColor` from `src/lib` to convert the image into
 //! a color pixel iterator. The simulator uses the `ColorOled` theme to display the image in color
-//!
-//! Note that this requires the `bmp` feature to be turned on for `embedded-graphics`. Turn it on
-//! with the following in `Cargo.toml`:
-//!
-//! [dependencies]
-//! embedded-graphics = { version = "*", features = [ "bmp" ] }
 
-use embedded_graphics::image::ImageBmp;
-use embedded_graphics::pixelcolor::Rgb565;
-use embedded_graphics::prelude::*;
+use embedded_graphics::{image::Image, pixelcolor::Rgb565, prelude::*};
 use embedded_graphics_simulator::{SimulatorDisplay, WindowBuilder};
+use tinybmp::Bmp;
 
 fn main() -> Result<(), core::convert::Infallible> {
-    let mut display = SimulatorDisplay::new(Size::new(128, 128));
+    let mut display: SimulatorDisplay<Rgb565> = SimulatorDisplay::new(Size::new(128, 128));
+
+    let bmp = Bmp::from_slice(include_bytes!("./rust-pride.bmp")).unwrap();
+
+    let image = Image::new(&bmp, Point::zero());
```

Comment:
IMO the examples should always annotate `Image` with a color type, because this defines the color format that's stored inside the image. The type inferred by the compiler equals the display color type, which might not be correct in all cases and can cause portability issues.

Reply:
Fair, which one is better do you think?

```rust
let image: Image<_, Rgb565> = Image::new(&bmp, Point::zero());
```

or 

```rust
let image: Image<Bmp, Rgb565> = Image::new(&bmp, Point::zero());
```

I'll go with the latter for now but let me know if that's a stupid idea.

Reply:
Both variants are OK and I don't have a preference.